examples/demo/src/products/ProductEdit.tsx

```diff
@@ -32,8 +35,21 @@ const RichTextInput = React.lazy(() =>
 );
 
 const ProductTitle = () => {
+    const appTitle = useDefaultTitle();
+    const translate = useTranslate();
     const record = useRecordContext<Product>();
-    return record ? <span>Poster "{record.reference}"</span> : null;
+    const getResourceLabel = useGetResourceLabel();
+
+    const pageTitle = translate('ra.page.edit', {
+        name: getResourceLabel('products', 1),
+        recordRepresentation: `"${record?.reference}"`,
+    });
```

Comment:
I did not use `usePageTitle` here because we want the quotes around the `name` in titles but not in the record representation